packages/core/src/odata/common/request/odata-request.ts

```diff
@@ -189,6 +180,27 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
     }
   }
 
+  /**
```

Comment:
I agree that the `customHeaders` were there twice and this should be fixed. But I do not like the current refactoring. `basicHeaders()` is not a good name for defaultHeader+etag+content-type+custom headers. I think the headers have nothing to do which each other and I would include them separately:

return{
   accept: 'application/json',
  ...destinationRelatedHeaders,
  ...csrfHeaders,
  ...this.getETagHeader()
  ...this.getContentTypeHeader()
  ...this.config.customHeaders
}

Reply:
Discussed in other PR.